[x86/Linux] Fix unknown type name 'PTR_RUNTIME_FUNCTION'

seanshpark opened the pull request:
Fix compile error for x86/Linux
- add PTR_RUNTIME_FUNCTION and others in daccess.h, clrnt.h and palrt.h
- these are copy of !FEATURE_PAL in corecompile.h

seanshpark commented:
@janvorli , @jkotas , @gkhanna79 , PTAL

seanshpark commented on src/pal/inc/pal.h:
```diff
@@ -4706,6 +4706,12 @@ typedef struct _RUNTIME_FUNCTION {
     DWORD UnwindData;
 } RUNTIME_FUNCTION, *PRUNTIME_FUNCTION;
 
+#if defined(_X86_)
```
OK, thank you :)

seanshpark commented:
If it is ok, I'd like to move `PTR_RUNTIME_FUNCTION`, `RUNTIME_FUNCTION__BeginAddress` and `RUNTIME_FUNCTION_INDIRECT` in `corcompile.h` also.

janvorli commented:
@seanshpark we need to have these in corcompile.h for Windows x86, since neither pal.h nor palrt.h are included for Windows builds. 
So thinking about it more, I would recommend reverting the change you did to corcompile.h before and only wrap the RUNTIME_FUNCTION definition in #ifndef FEATURE_PAL there and drop the current change completely.

seanshpark commented:
@janvorli , oh, it's not a good idea to change. OK then, could you review current changes? I hope these are what you meant.

janvorli commented:
@seanshpark I cannot see any new changes here - maybe you've forgotten to push?

seanshpark commented:
@janvorli , um... isn't current change https://github.com/dotnet/coreclr/pull/8272/commits/e943f060bf1cfcecc4e355d13146e5ced4689b80 has what you pointed in https://github.com/dotnet/coreclr/pull/8272#discussion_r89284212 ?

janvorli commented:
I've said in my next comment:
So thinking about it more, I would recommend reverting the change you did to corcompile.h before and only wrap the RUNTIME_FUNCTION definition in #ifndef FEATURE_PAL there (meaning in the corcompile.h) and **drop the current change completely**

seanshpark commented:
@janvorli , OK, got it. I am sorry for my misunderstanding. I've dropped previous adding defines and changed to your guide. Now the code is better with smaller changes from what it was. If this is also going in a wrong direction, please point out again.

seanshpark commented:
@dotnet-bot test Windows_NT x64 Debug Build and Test please
@dotnet-bot test Windows_NT x86 legacy_backend Checked Build and Test please

seanshpark commented:
@dotnet-bot test Windows_NT x86 compatjit Checked Build and Test please

seanshpark commented:
@dotnet-bot test Windows_NT x86 legacy_backend Checked Build and Test please
@dotnet-bot test Windows_NT x86 compatjit Checked Build and Test please

RussKeldorph commented:
Please ignore the x86 compatjit and x86 legacy_backend job failures.  They have been removed from future PRs.

jkotas commented on src/inc/corcompile.h:
```diff
@@ -71,7 +71,8 @@ typedef DPTR(struct CORCOMPILE_VIRTUAL_SECTION_INFO)
+#ifdef _TARGET_X86_
+#ifndef FEATURE_PAL
 //
 // x86 ABI does not define RUNTIME_FUNCTION. Define our own to allow unification between x86 and other platforms.
```
Make sure that the definition that we end up picking up from the PAL has the same two fields like this one - it does not appear to be the case.

jkotas replied:
We are not converged to the three field version even on non-x86 platforms today:

The only platform that uses the three field version is amd64 (on both Windows and Unix). 

arm and arm64 (on both Windows and Unix) use the two field version. Maybe that actually get compiled with the three field version on Unix, but the third field is just a dead weight. It is not used for anything. We should get rid of it.

janvorli replied:
Ok, then we should also modify the definition in PAL to ifdef out the third field for other than AMD64.

seanshpark replied:
Hope #8305 is the one.

seanshpark commented:
@dotnet-bot test Windows_NT x86 legacy_backend Checked Build and Test please
@dotnet-bot test Windows_NT x86 compatjit Checked Build and Test please

seanshpark commented:
Can this be merged? or do I have anything else to do?

janvorli commented:
@seanshpark sure, I am sorry, I have somehow confused myself into thinking that #8305 contained this chagne.

seanshpark commented:
Thank you :)
